src/components/AddRpcButton.tsx

```diff
@@ -74,16 +91,37 @@ function getErrorMessage(error: any): { errorMessage: string | null, isUserRejec
 
 export function AddRpcButton() {
   const { addRpcEndpoint } = useAddRpcEndpoint()
-  const [{ state, errorMessage }, setState] = useState<AddToWalletState>(DEFAULT_STATE)
+  const { isConnected, connect } = useConnect();
+  const { data: walletClient } = useWalletClient()
+  const [{ state, errorMessage, addOnceConnected }, setState] = useState<AddToWalletState>(DEFAULT_STATE)
   const isAdding = state === 'adding'
+  const isConnecting = state === 'connecting'
+  const isLoading = isAdding || isConnecting
+  const [addRpcPromise, setAddRpcPromise] = useState<Promise<boolean> | null>(null)
+
+  const handleError = useCallback((error) => {
+    const { errorMessage: message, isError, isUserRejection } = getErrorMessage(error)
+    isError && sendToAnalytics('error_add_rpc')
+    isUserRejection && sendToAnalytics('rejected_by_user_adding_rpc')
+
+    if (isError || isUserRejection) {
+      // Display the error
+      setState({ state: 'error', errorMessage: message, addOnceConnected: false })          
+    } else {
+      // Not an error: i.e The user is connecting
+      setState(DEFAULT_STATE)
+    }
+    setAddRpcPromise(null)
+  }, [setState])
   
 
   const addToWallet = useCallback(() => {
```

Comment:
Would be also great to extract the hook into separate file

Reply:
https://github.com/mevblocker/web/pull/64